### ISSUE TITLE
color-no-hex should optionally not report for variable assignment

### ISSUE THREAD

**Issue by mehcode** [stylelint/stylelint#1006]
Something like `ignore: ["variables"]`.

**Comment by jeddy3** [comment]
Would using the [special disable comments](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#turning-rules-on-and-off) not suffice here? e.g.

```css

/* stylelint-disable color-no-hex */

:root { --custom-property-name: #000; }
$scss-and-precss-variable-name: #000;
$scss-map: (name: #000);
@less-variable-name: #000;
stylus-variable-name = #000
...

/* stylelint-enable color-no-hex */

...
```

**Comment by mehcode** [comment]
You know.. I must have been tired. Thinking back I'm really looking for an `enforce-variable-usage` rule. I'd want it to apply to any color values that are not themselves variable assignment. Should I make another issue for that?

**Comment by jeddy3** [comment]
It's specific to *custom-properties* (e.g. `color: var(--custom-property-name)`), then please do create another issue for that. If it's for some non-standard syntax, then you'll need to create your own [plugin](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#plugins). There's [a page](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) in the developer guide that will help you get started. If you're specifically after scss-like syntax (e.g. scss and precss), then I think there's discussion about creating such a plugin [over here](https://github.com/kristerkari/stylelint-scss/issues/4).
